### PR TITLE
New payment processing policy

### PR DESCRIPTION
**Pull request by leogrim**
@andrewconner @eishay 
This PR depends one the other one.

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentProcessingCommander.scala [hunk 1]
```diff
@@ -44,9 +48,15 @@ class PaymentProcessingCommanderImpl @Inject() (
...
-  //if you owe us more than $100 we will charge your card even if your billing cycle is not up
+  private[payments] val MAX_BALANCE = DollarAmount.dollars(1000) //if you owe us more than $100 we will charge your card even if your billing cycle is not up
   private[payments] val MIN_BALANCE = DollarAmount.dollars(1) //if you are carrying a balance of less then one dollar you will not be charged (to much cost overhead)
+  private[payments] val PAYMENT_CYCLE = PaymentCycle.months(1) //how often oustanding balances will be charged
+
+  private def nextPaymentDueAt(account: PaidAccount): Option[DateTime] = {
+    val strictlyDueAt = clock.now() plusMonths PAYMENT_CYCLE.months
+    val looselyDueAt = strictlyDueAt plusDays 1
+    if (account.planRenewal isBefore looselyDueAt) None else Some(strictlyDueAt) // do not schedule a payment just before plan renewal
+  }
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentProcessingCommander.scala:59]: @andrewconner

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PlanRenewalCommander.scala [hunk 2]
```diff
@@ -50,11 +50,10 @@ class PlanRenewalCommanderImpl @Inject() (
...
         val plan = paidPlanRepo.get(account.planId)
         if (account.planRenewal isBefore now) {
-          val nextPlanRenewal = account.planRenewal plusMonths plan.billingCycle.month
+          val nextPlanRenewal = account.planRenewal plusMonths plan.billingCycle.months
           val fullCyclePrice = plan.pricePerCyclePerUser * account.activeUsers
-          val paymentDueAt = if (fullCyclePrice > DollarAmount.ZERO) Some(now) else account.paymentDueAt
           val renewedAccount = paidAccountRepo.save(
-            account.withReducedCredit(fullCyclePrice).withPlanRenewal(nextPlanRenewal).withPaymentDueAt(paymentDueAt)
+            account.withReducedCredit(fullCyclePrice).withPlanRenewal(nextPlanRenewal).withPaymentDueAt(Some(now))
           )
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PlanRenewalCommander.scala:57]: @andrewconner will just filter out `LowBalanceIgnored` from Slack and activity logs

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentProcessingCommander.scala (hunk 1 above)
Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentProcessingCommander.scala:59]: I assume if `nextPaymentDueAt` is `None`, it gets recaculated at `planRenewal`?

**Comment by andrewconner** [comment]
:+1: (Hard to tell what's a superset of the previous one, but things seem reasonable)

File: kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentProcessingCommander.scala (hunk 1 above)
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/payments/PaymentProcessingCommander.scala:59]: yep

**Pull request merged**